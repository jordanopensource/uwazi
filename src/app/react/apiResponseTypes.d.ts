import { UserGroupSchema } from 'shared/types/userGroupType';
import { UserSchema } from 'shared/types/userType';
import {
  PreserveConfig,
  Settings,
  SettingsFilterSchema,
  SettingsLinkSchema,
} from 'shared/types/settingsType';
import { LanguageSchema } from 'shared/types/commonTypes';
import { ThesaurusSchema, ThesaurusValueSchema } from 'shared/types/thesaurusType';
import { TemplateSchema } from 'shared/types/templateType';

export interface GroupMemberSchema {
  refId: string;
  username: string;
}

export interface ClientUserGroupSchema extends Omit<UserGroupSchema, '_id' | 'members'> {
  _id?: string;
  members: GroupMemberSchema[];
}

export interface ClientUserSchema extends Omit<UserSchema, '_id' | 'groups'> {
  _id?: string;
  groups?: {
    _id: string;
    name: string;
  }[];
}

export interface ClientSettingsFilterSchema extends Omit<SettingsFilterSchema, '_id'> {
  _id?: string;
}

export interface ClientSublink extends Omit<ClientSettingsLinkSchema, 'sublinks'> {}

export interface ClientSettingsLinkSchema extends Omit<SettingsLinkSchema, '_id' | 'sublinks'> {
  _id?: string;
  sublinks?: ClientSublink[];
}

export interface ClientPreserveConfig extends Omit<PreserveConfig, 'config'> {
  config: {
    token: string;
    template: string;
    user?: string;
  }[];
}

export interface ClientLanguageSchema extends Omit<LanguageSchema, '_id'> {
  _id?: string;
}

export interface Template extends TemplateSchema, Omit<TemplateSchema, '_id'> {
  _id: string;
}

export interface ClientSettings
  extends Omit<Settings, '_id | filters | links | features | languages'> {
  _id?: string;
  filters?: ClientSettingsFilterSchema[];
  languages?: ClientLanguageSchema[];
  links?: ClientSettingsLinkSchema[];
  features?: Omit<Settings['features'], 'preserve | metadataExtraction'> & {
    preserve?: ClientPreserveConfig;
    ocr?: { url: string };
    metadataExtraction?: {
      url: string;
      templates?: {
        template: string;
        properties: string[];
      }[];
    };
    paragraphExtraction?: boolean;
    [k: string]: unknown | undefined;
  };
}

export interface ClientThesaurus extends ThesaurusSchema, Omit<ThesaurusSchema, '_id | values'> {
  _id: ObjectIdSchema;
  values: ClientThesaurusValue[];
}

export interface ClientThesaurusValue extends Omit<ThesaurusValueSchema, '_id'> {
  id?: string;
  label: string;
  values?: {
    id?: string;
    label: string;
    name?: string;
  }[];
}

export interface ClientRelationshipType {
  _id: string;
  name: string;
}
