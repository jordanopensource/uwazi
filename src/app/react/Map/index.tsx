/* eslint-disable react/jsx-props-no-spreading */
import loadable from '@loadable/component';
import React from 'react';
import { Map, Layer } from 'app/Map/MapContainer';
import * as helper from './helper';

const LMap = loadable(async () => {
  const { LMap: LMapComponent } = await import(/* webpackChunkName: "LazyLoadMap" */ './LMap');
  return (props: any) => <LMapComponent {...props} />;
});

export { default as Markers } from './Markers';
export { helper, LMap, Map };
export type { Layer };
