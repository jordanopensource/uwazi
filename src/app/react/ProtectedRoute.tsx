import React, { ReactElement } from 'react';
import { Navigate, Outlet } from 'react-router-dom';
import { store } from 'app/store';
import { ClientSettings } from 'app/apiResponseTypes';

const ProtectedRoute = ({
  children,
  allowedRoles,
}: {
  children: ReactElement;
  allowedRoles?: string[];
}) => {
  const userId = store?.getState().user.get('_id');
  const userRole = store?.getState().user.get('role') || '';
  if (allowedRoles && allowedRoles.includes(userRole)) {
    return children || <Outlet />;
  }

  if (!allowedRoles && userId) {
    return children || <Outlet />;
  }

  return <Navigate to="/login" replace />;
};

const adminsOnlyRoute = (element: ReactElement) => (
  <ProtectedRoute allowedRoles={['admin']}>{element}</ProtectedRoute>
);

const privateRoute = (element: ReactElement, settings: ClientSettings | undefined) =>
  !settings?.private ? element : <ProtectedRoute>{element}</ProtectedRoute>;

const loggedInUsersRoute = (element: ReactElement) => <ProtectedRoute>{element}</ProtectedRoute>;
export { loggedInUsersRoute, adminsOnlyRoute, privateRoute, ProtectedRoute };
