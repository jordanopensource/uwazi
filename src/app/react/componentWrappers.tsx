/* eslint-disable comma-spacing */
/* eslint-disable react/no-multi-comp */
import React, { useContext } from 'react';
import {
  useLocation,
  useMatches,
  useNavigate,
  useOutlet,
  useParams,
  useSearchParams,
} from 'react-router-dom';
import { AppMainContext } from './App/AppMainContext';
import { searchParamsFromSearchParams } from './utils/routeHelpers';

const withRouter =
  <T,>(Component: React.ComponentClass<T & { params?: any }, any>) =>
  (props: T & { params?: any }) => {
    const location = useLocation();
    const navigate = useNavigate();
    const params = useParams();
    const matches = useMatches();
    const [searchParams, setSearchParams] = useSearchParams();
    const { q: searchString } = searchParamsFromSearchParams(searchParams);
    const paramsFromProps = { ...props.params, q: searchString || props.params?.q };

    return (
      <Component
        {...props}
        navigate={navigate}
        location={location}
        params={{ ...params, ...(paramsFromProps || {}) }}
        matches={matches}
        searchParams={searchParams}
        setSearchParams={setSearchParams}
      />
    );
  };

const withContext =
  <T,>(Component: React.ComponentClass<T, any>) =>
  (props: T) => {
    const mainContext = useContext(AppMainContext);
    return <Component {...props} mainContext={mainContext} />;
  };

const withOutlet =
  <T,>(Component: React.ComponentClass<T, any>) =>
  (props: T) => {
    const outlet = useOutlet(AppMainContext);
    return <Component {...props} outlet={outlet} />;
  };

export { withRouter, withContext, withOutlet };
