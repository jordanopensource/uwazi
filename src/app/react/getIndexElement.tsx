import React from 'react';
import { Navigate } from 'react-router-dom';
import { ClientSettings } from 'app/apiResponseTypes';
import { validateHomePageRoute } from './utils/routeHelpers';
import { PageView } from './Pages/PageView';
import { LibraryTable } from './Library/LibraryTable';
import { LibraryMap } from './Library/LibraryMap';
import { LibraryCards } from './Library/Library';
import { Login } from './Users/Login';
import { ViewerRoute } from './Viewer/ViewerRoute';

const deconstructSearchQuery = (query?: string) => {
  if (!query) return '';
  if (query.startsWith('?q=')) {
    return decodeURI(query.substring(1).split('=')[1]);
  }
  return `(${query.substring(1)})`;
};

const getCustomLibraryPage = (customHomePage: string[]) => {
  const [query] = customHomePage.filter(path => path.startsWith('?'));
  const searchQuery = deconstructSearchQuery(query);
  const queryString = query ? searchQuery : '';

  if (customHomePage.includes('map')) {
    return <LibraryMap params={{ q: queryString }} />;
  }

  if (customHomePage.includes('table')) {
    return <LibraryTable params={{ q: queryString }} />;
  }

  return <LibraryCards params={{ q: queryString }} />;
};

const getLibraryDefault = (
  userId: string | undefined,
  defaultLibraryView: string | undefined,
  privateInstance: boolean | undefined
) => {
  if (userId) {
    return <Navigate to="/library/?q=(includeUnpublished:!t)" state={{ isClient: true }} />;
  }

  if (privateInstance) {
    return <Login />;
  }

  switch (defaultLibraryView) {
    case 'table':
      return <LibraryTable />;

    case 'map':
      return <LibraryMap />;

    case 'cards':
    default:
      return <LibraryCards />;
  }
};

// eslint-disable-next-line max-statements
const getIndexElement = (settings: ClientSettings | undefined, userId: string | undefined) => {
  const customHomePage = settings?.home_page ? settings?.home_page.split('/').filter(v => v) : [];
  const isValidHomePage = validateHomePageRoute(settings?.home_page || '');
  let element = <Navigate to={customHomePage.join('/')} />;
  let parameters;

  switch (true) {
    case !isValidHomePage || customHomePage.length === 0:
      element = getLibraryDefault(userId, settings?.defaultLibraryView, settings?.private);
      break;

    case isValidHomePage && customHomePage.includes('page'):
      {
        const pageId = customHomePage[customHomePage.indexOf('page') + 1];
        element = <PageView params={{ sharedId: pageId }} />;
        parameters = { sharedId: pageId };
      }
      break;

    case isValidHomePage && customHomePage.includes('entity'):
      {
        const pageId = customHomePage[customHomePage.indexOf('entity') + 1];
        element = <ViewerRoute params={{ sharedId: pageId }} />;
      }
      break;

    case isValidHomePage && customHomePage.includes('library'):
      element = getCustomLibraryPage(customHomePage);
      break;

    default:
      break;
  }

  return { element, parameters };
};

export { getIndexElement };
