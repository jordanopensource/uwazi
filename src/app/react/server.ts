import { Application } from 'express';
import { EntryServer } from './entry-server';

export const serverSideRender = (app: Application) => {
  app.get(/^\/(?!api(\/|$)).*$/, EntryServer);
};
