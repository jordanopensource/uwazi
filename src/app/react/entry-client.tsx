import React from 'react';
import { hydrateRoot } from 'react-dom/client';
import * as Sentry from '@sentry/react';
import {
  RouterProvider,
  createBrowserRouter,
  createRoutesFromChildren,
  matchRoutes,
  useLocation,
  useNavigationType,
} from 'react-router-dom';
import { Provider } from 'jotai';
import { Provider as ReduxProvider } from 'react-redux';
import type { RequestError } from 'V2/shared/errorUtils';
import { ErrorBoundary } from './V2/Components/ErrorHandling';
import './App/sockets';
import { getRoutes } from './Routes';
import CustomProvider from './App/Provider';
import { settingsAtom, atomStore, userAtom } from './V2/atoms';
import { store } from './store';

declare global {
  interface Window {
    __loadingError__?: RequestError;
  }
}

if (window.SENTRY_APP_DSN) {
  Sentry.init({
    release: window.UWAZI_VERSION,
    environment: window.UWAZI_ENVIRONMENT,
    dsn: window.SENTRY_APP_DSN,
    integrations: [
      new Sentry.BrowserTracing({
        routingInstrumentation: Sentry.reactRouterV6Instrumentation(
          React.useEffect,
          useLocation,
          useNavigationType,
          createRoutesFromChildren,
          matchRoutes
        ),
      }),
      new Sentry.Replay(),
    ],

    tracesSampleRate: 0.1,
  });
}

const router = createBrowserRouter(
  getRoutes(atomStore.get(settingsAtom), atomStore.get(userAtom)?._id)
);

const App = () => (
  <ReduxProvider store={store as any}>
    <CustomProvider>
      <Provider store={atomStore}>
        <ErrorBoundary>
          <RouterProvider router={router} fallbackElement={null} />
        </ErrorBoundary>
      </Provider>
    </CustomProvider>
  </ReduxProvider>
);

const container = document.getElementById('root');
const root = window.__loadingError__ === undefined ? hydrateRoot(container!, <App />) : container;
const silentWarnings = [
  'Warning: %s uses the legacy childContextTypes API which is no longer supported and will be removed in the next major release.',
  'Warning: %s: Support for defaultProps will be removed from function components in a future major release. Use JavaScript default parameters instead.%s',
  'Warning: %s uses the legacy contextTypes API which is no longer supported and will be removed in the next major release.',
  'Warning: findDOMNode is deprecated and will be removed in the next major release.',
];

const isSilentWarning = (warning: any) =>
  silentWarnings.find(w => typeof warning === 'string' && warning.includes(w)) !== undefined;

const origConsoleError = window.console.error;

window.console.error = (...args) => {
  if (isSilentWarning(args[0])) {
    return;
  }
  origConsoleError.apply(window.console, args);
};

export { root };
