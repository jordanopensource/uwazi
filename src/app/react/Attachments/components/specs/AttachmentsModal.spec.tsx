/**
 * @jest-environment jsdom
 */
import React from 'react';
import { fireEvent, screen } from '@testing-library/react';
import { defaultState, renderConnectedContainer } from 'app/utils/test/renderConnected';
import { AttachmentsModalCmp, AttachmentsModalProps } from '../AttachmentsModal';

describe('Attachments Modal', () => {
  let props: AttachmentsModalProps;

  beforeEach(() => {
    props = {
      entitySharedId: '123',
      storeKey: '456',
      onClose: jasmine.createSpy('onClose'),
      uploadAttachmentFromUrl: jasmine.createSpy('uploadAttachmentFromUrl'),
      isOpen: true,
      uploadAttachment: jasmine.createSpy('uploadAttachment'),
      model: '',
    };
  });

  const render = (otherProps = {}) => {
    const renderProps = { ...props, ...otherProps };
    renderConnectedContainer(<AttachmentsModalCmp {...renderProps} />, () => defaultState);
  };

  it('Should open the modal according to isOpen property value', () => {
    render({ isOpen: false });
    expect(screen.queryByRole('heading')).not.toBeInTheDocument();
    render({ isOpen: true });
    expect(screen.queryAllByRole('heading').length).toBe(2);
  });

  it('Should match render of upload form', () => {
    render({ isOpen: true });
    expect(screen.getByRole('dialog')).toMatchSnapshot();
  });

  describe('Attachment from web', () => {
    beforeEach(() => {
      render({ isOpen: true });
      const webTab = screen.getByRole('tab', { name: 'Add from web' });
      fireEvent.click(webTab);
    });

    it('Should match render of web form', () => {
      expect(screen.getByRole('dialog')).toMatchSnapshot();
    });

    it('Should submit web form', () => {
      render();
      const urlInput = screen.getAllByRole('textbox');
      const attachmentData = { url: 'http://test.test', name: 'testName' };

      fireEvent.change(urlInput.at(0)!, { target: { value: attachmentData.url } });
      fireEvent.change(urlInput.at(1)!, { target: { value: attachmentData.name } });

      fireEvent.click(screen.getByRole('button', { name: 'Add from URL' }));
      expect(props.uploadAttachmentFromUrl).toHaveBeenCalledWith(
        props.entitySharedId,
        attachmentData,
        {
          __reducerKey: props.storeKey,
          model: '',
        }
      );
    });
  });

  it('Should call onClose', () => {
    render({ isOpen: true });
    fireEvent.click(screen.getByRole('button', { name: 'Cancel' }));
    expect(props.onClose).toHaveBeenCalled();
  });
});
