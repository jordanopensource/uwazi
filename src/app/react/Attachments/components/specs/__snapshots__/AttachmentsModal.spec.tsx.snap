// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Attachments Modal Attachment from web Should match render of web form 1`] = `
<div
  aria-modal="true"
  class="ReactModal__Content attachments-modal"
  role="dialog"
  tabindex="-1"
>
  <div
    class="attachments-modal__header"
  >
    <h4>
      <span
        class="translation active"
      >
        Supporting files
      </span>
    </h4>
    <button
      class="attachments-modal__close"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-times fa-w-11 "
        data-icon="times"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 352 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
          fill="currentColor"
        />
      </svg>
      <span>
        <span
          class="translation active"
        >
          Cancel
        </span>
      </span>
    </button>
  </div>
  <div
    class="attachments-modal__content"
  >
    <div>
      <div
        class="attachments-modal__tabs"
      >
        <div
          aria-controls="tabpanel-uploadComputer"
          aria-selected="false"
          class="tab-link modal-tab-1"
          component="div"
          id="tab-uploadComputer"
          role="tab"
          style=""
        >
          <span
            class="translation active"
          >
            Upload from computer
          </span>
        </div>
        <div
          aria-controls="tabpanel-uploadWeb"
          aria-selected="true"
          class="tab-link modal-tab-2 tab-link-active"
          component="div"
          id="tab-uploadWeb"
          role="tab"
          style="font-weight: bold;"
        >
          <span
            class="translation active"
          >
            Add from web
          </span>
        </div>
      </div>
      <div
        class="attachments-modal__tabs-content"
      >
        <div
          aria-labelledby="tab-uploadComputer"
          class="tab-content centered"
          id="tabpanel-uploadComputer"
          role="tabpanel"
          style="display: none;"
        />
        <div
          aria-labelledby="tab-uploadWeb"
          class="tab-content centered tab-content-visible"
          id="tabpanel-uploadWeb"
          role="tabpanel"
          style=""
        >
          <div
            class="wrapper-web"
          >
            <form
              class=""
            >
              <div
                class="form-group has-feedback "
              >
                <div>
                  <input
                    class="form-control web-attachment-url"
                    name="urlForm.url"
                    placeholder="Paste URL here"
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <div
                class="form-group form-group "
              >
                <div
                  class="field"
                >
                  <input
                    class="form-control web-attachment-name"
                    name="urlForm.name"
                    placeholder="Title"
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <button
                class="btn"
                type="submit"
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-link fa-w-16 "
                  data-icon="link"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M326.612 185.391c59.747 59.809 58.927 155.698.36 214.59-.11.12-.24.25-.36.37l-67.2 67.2c-59.27 59.27-155.699 59.262-214.96 0-59.27-59.26-59.27-155.7 0-214.96l37.106-37.106c9.84-9.84 26.786-3.3 27.294 10.606.648 17.722 3.826 35.527 9.69 52.721 1.986 5.822.567 12.262-3.783 16.612l-13.087 13.087c-28.026 28.026-28.905 73.66-1.155 101.96 28.024 28.579 74.086 28.749 102.325.51l67.2-67.19c28.191-28.191 28.073-73.757 0-101.83-3.701-3.694-7.429-6.564-10.341-8.569a16.037 16.037 0 0 1-6.947-12.606c-.396-10.567 3.348-21.456 11.698-29.806l21.054-21.055c5.521-5.521 14.182-6.199 20.584-1.731a152.482 152.482 0 0 1 20.522 17.197zM467.547 44.449c-59.261-59.262-155.69-59.27-214.96 0l-67.2 67.2c-.12.12-.25.25-.36.37-58.566 58.892-59.387 154.781.36 214.59a152.454 152.454 0 0 0 20.521 17.196c6.402 4.468 15.064 3.789 20.584-1.731l21.054-21.055c8.35-8.35 12.094-19.239 11.698-29.806a16.037 16.037 0 0 0-6.947-12.606c-2.912-2.005-6.64-4.875-10.341-8.569-28.073-28.073-28.191-73.639 0-101.83l67.2-67.19c28.239-28.239 74.3-28.069 102.325.51 27.75 28.3 26.872 73.934-1.155 101.96l-13.087 13.087c-4.35 4.35-5.769 10.79-3.783 16.612 5.864 17.194 9.042 34.999 9.69 52.721.509 13.906 17.454 20.446 27.294 10.606l37.106-37.106c59.271-59.259 59.271-155.699.001-214.959z"
                    fill="currentColor"
                  />
                </svg>
                Â  
                <span
                  class="translation active"
                >
                  Add from URL
                </span>
              </button>
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Attachments Modal Should match render of upload form 1`] = `
<div
  aria-modal="true"
  class="ReactModal__Content attachments-modal"
  role="dialog"
  tabindex="-1"
>
  <div
    class="attachments-modal__header"
  >
    <h4>
      <span
        class="translation active"
      >
        Supporting files
      </span>
    </h4>
    <button
      class="attachments-modal__close"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-times fa-w-11 "
        data-icon="times"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 352 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
          fill="currentColor"
        />
      </svg>
      <span>
        <span
          class="translation active"
        >
          Cancel
        </span>
      </span>
    </button>
  </div>
  <div
    class="attachments-modal__content"
  >
    <div>
      <div
        class="attachments-modal__tabs"
      >
        <div
          aria-controls="tabpanel-uploadComputer"
          aria-selected="true"
          class="tab-link modal-tab-1 tab-link-active"
          component="div"
          id="tab-uploadComputer"
          role="tab"
          style="font-weight: bold;"
        >
          <span
            class="translation active"
          >
            Upload from computer
          </span>
        </div>
        <div
          aria-controls="tabpanel-uploadWeb"
          aria-selected="false"
          class="tab-link modal-tab-2"
          component="div"
          id="tab-uploadWeb"
          role="tab"
        >
          <span
            class="translation active"
          >
            Add from web
          </span>
        </div>
      </div>
      <div
        class="attachments-modal__tabs-content"
      >
        <div
          aria-labelledby="tab-uploadComputer"
          class="tab-content centered tab-content-visible"
          id="tabpanel-uploadComputer"
          role="tabpanel"
        >
          <div
            class="attachments-modal__dropzone"
            role="presentation"
            tabindex="0"
          >
            <button
              class="btn"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-link fa-w-16 "
                data-icon="link"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M326.612 185.391c59.747 59.809 58.927 155.698.36 214.59-.11.12-.24.25-.36.37l-67.2 67.2c-59.27 59.27-155.699 59.262-214.96 0-59.27-59.26-59.27-155.7 0-214.96l37.106-37.106c9.84-9.84 26.786-3.3 27.294 10.606.648 17.722 3.826 35.527 9.69 52.721 1.986 5.822.567 12.262-3.783 16.612l-13.087 13.087c-28.026 28.026-28.905 73.66-1.155 101.96 28.024 28.579 74.086 28.749 102.325.51l67.2-67.19c28.191-28.191 28.073-73.757 0-101.83-3.701-3.694-7.429-6.564-10.341-8.569a16.037 16.037 0 0 1-6.947-12.606c-.396-10.567 3.348-21.456 11.698-29.806l21.054-21.055c5.521-5.521 14.182-6.199 20.584-1.731a152.482 152.482 0 0 1 20.522 17.197zM467.547 44.449c-59.261-59.262-155.69-59.27-214.96 0l-67.2 67.2c-.12.12-.25.25-.36.37-58.566 58.892-59.387 154.781.36 214.59a152.454 152.454 0 0 0 20.521 17.196c6.402 4.468 15.064 3.789 20.584-1.731l21.054-21.055c8.35-8.35 12.094-19.239 11.698-29.806a16.037 16.037 0 0 0-6.947-12.606c-2.912-2.005-6.64-4.875-10.341-8.569-28.073-28.073-28.191-73.639 0-101.83l67.2-67.19c28.239-28.239 74.3-28.069 102.325.51 27.75 28.3 26.872 73.934-1.155 101.96l-13.087 13.087c-4.35 4.35-5.769 10.79-3.783 16.612 5.864 17.194 9.042 34.999 9.69 52.721.509 13.906 17.454 20.446 27.294 10.606l37.106-37.106c59.271-59.259 59.271-155.699.001-214.959z"
                  fill="currentColor"
                />
              </svg>
              Â  
              <span
                class="translation active"
              >
                Upload and select file
              </span>
            </button>
            <input
              aria-label="fileInput"
              style="display: none;"
              type="file"
            />
            <h4
              class="attachments-modal__dropzone-title"
            >
              <span
                class="translation active"
              >
                Drag and drop file in this window to upload
              </span>
            </h4>
          </div>
        </div>
        <div
          aria-labelledby="tab-uploadWeb"
          class="tab-content centered"
          id="tabpanel-uploadWeb"
          role="tabpanel"
          style="display: none;"
        />
      </div>
    </div>
  </div>
</div>
`;
